common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -465,7 +470,11 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
      * consumer (one thread!).
      */
     public static <T> Queue<T> newMpscQueue() {
-        return new MpscLinkedQueue<T>();
+        if (hasUnsafe()) {
+            return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();
```

Comment:
@johnou @normanmaurer chunked queue will be good for the case that you will grow to a finite/controllable size, for fixed cases I believe ArrayQueue will still be faster, I'll use chunked queue later in my projects but scenarios for it have to arise first.